### PR TITLE
url in docstrings for modules   string - statistic - specialfns

### PR DESCRIPTION
**Pull request by mmatera**
Here we go with the first bunch from #668. There are 14 in total, covering three submodules.

### CODE REVIEW

**Comment by rocky** [comment]
> Here we go with the first bunch from #668. There are 14 in total, covering three submodules.

Thanks - much appreciated. Give me a day or so to look at these.

**Comment by mmatera** [comment]
> > Here we go with the first bunch from #668. There are 14 in total, covering three submodules.
> 
> Thanks - much appreciated. Give me a day or so to look at these.

Take your time. Just do not merge things over mathics.builtin before finishing this :).

File: mathics/builtin/specialfns/bessel.py [hunk 1]
```diff
@@ -737,7 +760,10 @@ class SphericalHankelH2(_Bessel):
 
 class StruveH(_Bessel):
     """
-    <url>:Struve functions H: https://en.wikipedia.org/wiki/Struve_function</url>
+
+    <url>:Struve functions H: https://en.wikipedia.org/wiki/Struve_function</url>\
+    (<url>:WMA link:https://reference.wolfram.com/language/ref/Struve.html</url>)
```
Comment (rocky) [inline mathics/builtin/specialfns/bessel.py:765]: In link Struve.html -> Struve.html

File: mathics/builtin/specialfns/elliptic.py [hunk 2]
```diff
@@ -72,6 +74,8 @@ def apply_phi_m(self, phi, m, evaluation):
 
 class EllipticF(SympyFunction):
     """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/EllipticF.html</url>
+
     <dl>
       <dt>'EllipticF[$m$]'
```
Comment (rocky) [inline mathics/builtin/specialfns/elliptic.py:80]: This function signature is wrong. It should be `EllipticF[$phi$, $m$]'

File: mathics/builtin/specialfns/expintegral.py [hunk 3]
```diff
@@ -10,6 +10,8 @@
 
 class ExpIntegralE(_MPMathFunction):
     """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/ExpIntegralE.html</url>
+
     <dl>
     <dt>'ExpIntegralE[$n$, $z$]'
```
Comment (rocky) [inline mathics/builtin/specialfns/expintegral.py:16]: Dt indentation.

File: mathics/builtin/specialfns/expintegral.py [hunk 4]
```diff
@@ -43,6 +47,8 @@ class ExpIntegralEi(_MPMathFunction):
 
 class ProductLog(_MPMathFunction):
     """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/ProductLog.html</url>
+
     <dl>
     <dt>'ProductLog[$z$]'
```
Comment (rocky) [inline mathics/builtin/specialfns/expintegral.py:53]: dl indentgation.

File: mathics/builtin/specialfns/expintegral.py (hunk 3 above)
Comment (rocky) [reply mathics/builtin/specialfns/expintegral.py:16]: Some other problem is causing these to not show up in index or when searching.

File: mathics/builtin/specialfns/gamma.py [hunk 5]
```diff
@@ -178,6 +182,8 @@ class Factorial(PostfixOperator, _MPMathFunction):
 
 class Factorial2(PostfixOperator, _MPMathFunction):
     """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/Factorial2.html</url>
```
Comment (rocky) [inline mathics/builtin/specialfns/gamma.py:185]: FYI Wiki lnk is https://en.wikipedia.org/wiki/Factorial

File: mathics/builtin/specialfns/gamma.py [hunk 6]
```diff
@@ -261,6 +267,8 @@ def fact2_generic(x):
 
 class Gamma(_MPMathMultiFunction):
     """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/Gamma.html</url>
```
Comment (rocky) [inline mathics/builtin/specialfns/gamma.py:270]: mpmath link is https://mpmath.org/doc/current/functions/gamma.html?#gamma
Sympy  link is https://mpmath.org/doc/current/functions/gamma.html?#gamma

File: mathics/builtin/string/characters.py [hunk 7]
```diff
@@ -182,6 +194,8 @@ def apply(self, s, evaluation):
 
 class ToUpperCase(Builtin):
     """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/ToUpperCase.html</url>
+
     <dl>
     <dt>'ToUpperCase[$s$]'
```
Comment (rocky) [inline mathics/builtin/string/characters.py:200]: dt indentation.

File: mathics/builtin/string/characters.py [hunk 8]
```diff
@@ -201,6 +215,8 @@ def apply(self, s, evaluation):
 
 class UpperCaseQ(Test):
     """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/UpperCaseQ.html</url>
+
     <dl>
     <dt>'UpperCaseQ[$s$]'
```
Comment (rocky) [inline mathics/builtin/string/characters.py:221]: dt indentation

File: mathics/builtin/statistics/orderstats.py [hunk 9]
```diff
@@ -145,6 +146,7 @@ def ranked(i):
 
 class Quartiles(Builtin):
     """
+    <url>:Quartile: https://en.wikipedia.org/wiki/Quartile</url> (<url>:WMA: https://reference.wolfram.com/language/ref/Quartiles.html</url>)
```
Comment (rocky) [inline mathics/builtin/statistics/orderstats.py:149]: Somes we have WMA and sometimes WMA link.

File: mathics/builtin/statistics/orderstats.py [hunk 10]
```diff
@@ -316,6 +326,8 @@ def apply(self, element, n, evaluation, options):
 
 class TakeSmallest(_RankedTakeSmallest):
     """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/TakeSmallest.html</url>
+
     <dl>
       <dt>'TakeSmallest[$list$, $f$, $n$]'
```
Comment (rocky) [inline mathics/builtin/statistics/orderstats.py:332]: $f$ does not seem to appear.

File: mathics/builtin/statistics/orderstats.py (hunk 9 above)
Comment (mmatera) [reply mathics/builtin/statistics/orderstats.py:149]: I tried to follow the existing pattern, but sometimes the pattern was not there...

File: mathics/builtin/string/patterns.py [hunk 11]
```diff
@@ -61,8 +63,10 @@ class DigitCharacter(Builtin):
 
 class EndOfLine(Builtin):
     r"""
+    <url>:WMA link:https://reference.wolfram.com/language/ref/EndOfLine.html</url>
+
     <dl>
-    <dt>'EndOfString'
+    <dt>'EndOfLine'
```
Comment (rocky) [inline mathics/builtin/string/patterns.py:69]: dt indentation

**Comment by rocky** [comment]
@mmatera I will be making the suggested changes now, and then I'll merge when that's done. 

In going over this it looks like in these files we'll have to do more iterations.

File: mathics/builtin/statistics/orderstats.py (hunk 9 above)
Comment (rocky) [reply mathics/builtin/statistics/orderstats.py:149]: > I tried to follow the existing pattern, but sometimes the pattern was not there...

Here's a pattern to follow. If WMA is not in parenthesis add "link", if it is in parenthesis as it is here, no "link" added. 

So this one is okay. But I think I have seen "WMA link" in parenthesis elsewhere and it is something to go over for another iteration.

**Comment by mmatera** [comment]
> @mmatera I will be making the suggested changes now, and then I'll merge when that's done.
> 
> (If I am going to make a pass over looking at this, and I feel I should, then I may as well correct those things that I see, even, although there is way more to do here. )
> 
> In going over this it looks like in these files we'll have to do more iterations.

OK. I was trying to fix just the urls, but sure, go ahead.

**Comment by rocky** [comment]
After CI finishes let's merge this. Probably tomorrow I will go over the PR that covers builtins bytearray, drawing, and list because there are bytearray and numpy fixes in https://github.com/Mathics3/mathics-core/tree/ByteArray-box

**Comment by mmatera** [comment]
> After CI finishes let's merge this. Probably tomorrow I will go over the PR that covers builtins bytearray, drawing, and list because there are bytearray and numpy fixes in https://github.com/Mathics3/mathics-core/tree/ByteArray-box

Maybe the best would be pick the involved files and put them in a separated PR.

**Comment by rocky** [comment]
> > After CI finishes let's merge this. Probably tomorrow I will go over the PR that covers builtins bytearray, drawing, and list because there are bytearray and numpy fixes in https://github.com/Mathics3/mathics-core/tree/ByteArray-box
> 
> Maybe the best would be pick the involved files and put them in a separated PR.

I think I am going to take you up on this kind offer #677 contains `builtins/bytearray.py`.  After this land's I will pull rebase the existing PRs, and put in another PR for `drawing/images`, after that lands I can rebase again, 
then put in numpy and bytearray fixes.

**Pull request merged**